src/goto-instrument/contracts/contracts.cpp

```diff
@@ -861,7 +861,7 @@ bool code_contractst::check_frame_conditions_function(const irep_idt &function)
   for(const auto &sym_pair : symbol_table)
   {
     const auto &sym = sym_pair.second;
-    if(sym.is_static_lifetime && sym.location.get_function() == function)
+    if(sym.is_static_lifetime && !sym.location.get_function().empty())
```

Comment:
Wouldn't this create CARs for ALL static local variables, like beyond the current function on which contracts are checked and its subfunction calls?
It might have a significant perf impact if lots of unrelated functions also introduce static local variables.

Reply:
It might be worth fixing the inlining routine instead. Inlined static locals should just behave as static locals within the caller. It seems weird and inconsistent to have the function body inlined into a caller but be mutating a static local that's tagged with a different function name.

Reply:
Indeed. 
Since all sub-function calls are inlined for contract checking, a better solution would be to detect static locals used in the body of the function under analysis rather than at the symbol table level.